script.js

```diff
@@ -1,12 +1,156 @@
-//You can edit ALL of the code here
+//Get DOM elements
+const rootHtml = document.getElementById("root"); // Get the root element where we'll add the episode cards
+const searchInputHtml = document.getElementById("search-input");
+const searchResultDisplayHtml = document.querySelector(".search-display");
+const episodeSelectHTML = document.querySelector("#select-html");
+
+// Global variables
+let allEpisodes;
+
+// Fetch episodes function
```

Comment:
In your fetch function you simply forgot to assign your result to allEpisodes array, hence reference error when filtering :)

Reply:
Good spot Maks!